src/librustc_mir/hair/pattern/check_match.rs

```diff
@@ -527,53 +527,61 @@ fn check_legality_of_move_bindings(cx: &MatchVisitor,
-
-        // x @ Foo(..) is legal, but x @ Foo(y) isn't.
-        if sub.map_or(false, |p| p.contains_bindings()) {
-            struct_span_err!(cx.tcx.sess, p.span, E0007,
-                             "cannot bind by-move with sub-bindings")
-                .span_label(p.span, "binds an already bound by-move value by moving it")
-                .emit();
-        } else if has_guard {
-            struct_span_err!(cx.tcx.sess, p.span, E0008,
-                      "cannot bind by-move into a pattern guard")
-                .span_label(p.span, "moves value into pattern guard")
-                .emit();
-        } else if let Some(by_ref_span) = by_ref_span {
-            struct_span_err!(
-                cx.tcx.sess,
-                p.span,
-                E0009,
-                "cannot bind by-move and by-ref in the same pattern",
-            )
-            .span_label(p.span, "by-move pattern here")
-            .span_label(by_ref_span, "both by-ref and by-move used")
-            .emit();
-        }
-    };
-
-    for pat in pats {
-        pat.walk(|p| {
-            if let PatKind::Binding(_, _, _, ref sub) = p.node {
-                if let Some(&bm) = cx.tables.pat_binding_modes().get(p.hir_id) {
-                    match bm {
-                        ty::BindByValue(..) => {
-                            let pat_ty = cx.tables.node_id_to_type(p.hir_id);
-                            if pat_ty.moves_by_default(cx.tcx, cx.param_env, pat.span) {
-                                check_move(p, sub.as_ref().map(|p| &**p));
+        for pat in pats {
+            pat.walk(|p| {
+                if let PatKind::Binding(_, _, _, ref sub) = p.node {
+                    if let Some(&bm) = cx.tables.pat_binding_modes().get(p.hir_id) {
+                        match bm {
+                            ty::BindByValue(..) => {
+                                let pat_ty = cx.tables.node_id_to_type(p.hir_id);
+                                if pat_ty.moves_by_default(cx.tcx, cx.param_env, pat.span) {
+                                    check_move(p, sub.as_ref().map(|p| &**p));
+                                }
                             }
+                            _ => {}
                         }
-                        _ => {}
+                    } else {
+                        cx.tcx.sess.delay_span_bug(pat.span, "missing binding mode");
                     }
-                } else {
-                    cx.tcx.sess.delay_span_bug(pat.span, "missing binding mode");
                 }
-            }
-            true
-        });
+                true
+            });
+        }
+    }
+    if !span_vec.is_empty(){
+        let span = MultiSpan::from_spans(span_vec.clone());
+        let mut err = struct_span_err!(
+                    cx.tcx.sess,
+                    span,
+                    E0009,
+                    "cannot bind by-move and by-ref in the same pattern",
+                );
```

Comment:
Nitpick: the appropriate indentation for this macro call is:

```
        let mut err = struct_span_err!(
            cx.tcx.sess,
            span,
            E0009,
            "cannot bind by-move and by-ref in the same pattern",
        );
```

Reply:
done!